tokens/contracts/erc-1155/ERC1155Lazy.sol

```diff
@@ -54,7 +55,7 @@ abstract contract ERC1155Lazy is IERC1155LazyMint, ERC1155BaseURI, Mint1155Valid
         }
     }
 
-    function mintAndTransfer(LibERC1155LazyMint.Mint1155Data memory data, address to, uint256 _amount) public override virtual {
+    function mintAndTransfer(LibERC1155LazyMint.Mint1155Data memory data, address to, uint256 _amount) public override virtual validateMinter {
         address minter = address(data.tokenId >> 96);
```

Comment:
The minter is this function is not `_msgSender()` but `address(data.tokenId >> 96)`.

The modifier will not work. Maybe replace the modifier by its require to give more flexibility?

Reply:
I think the same issue apply to ERC721Lazy

Reply:
This is related to the lazy minting process where Alice signs the NFT payload, but Bob sent it later on. In that case, Bob is the sender and mint for Alice.

In that case, who should be authorized? Bob because he's the one pushing the creation or Alice because she's the one that initially creates it? Both?

It feels more natural that Alice is the one that should be authorized and in that case yes this needs to be changed. If Bob should be the one authorized then it's valid